packages/captp/src/trap.js

```diff
@@ -0,0 +1,87 @@
+// @ts-check
+// Lifted mostly from `@agoric/eventual-send/src/E.js`.
+
+import './types';
+
+/**
+ * Default implementation of Trap for near objects.
+ *
+ * @type {TrapImpl}
+ */
+export const nearTrapImpl = harden({
+  applyFunction(target, args) {
+    return target(...args);
+  },
+  applyMethod(target, prop, args) {
+    return target[prop](...args);
+  },
+  get(target, prop) {
+    return target[prop];
+  },
+});
+
+const readOnlyProxyHandler = {
+  set(_target, _prop, _value) {
+    return false;
+  },
+  isExtensible(_target) {
+    return false;
```

Comment:
By returning isExtensible false here, and by allowing ownKeys to default to the empty target, the invariant enforcement may prevent an observed proxy from then claiming to have more properties than exist on the target.

Reply:
Inverting your comment, I've ensured that the proxy refuses all modifications, but doesn't report itself an non-extensible.